File: basics_of_computer_algorithms/chapter_one/EX1_1/js/EXT1_1.js
const readline = require('readline');

let A, B, C, D;

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

new Promise( (resolve) => { 
    rl.question('A? ', (entered_A) => {
        A =  parseFloat(entered_A);
        rl.question('B? ', (entered_B) => {
            B =  parseFloat(entered_B);
            rl.question('C? ', (entered_C) => {
                C =  parseFloat(entered_C);
                resolve();
                rl.close();
            });
        });
    });

}).then( ()=> {
    D = B*B - 4*A*C;
}).then( ()=> {
    if( D == 0) console.log(`Equation have one solution: ${-B/(2*A)}`);
    else if ( D > 0) console.log(`Equation have two solutions: ${(-B + D)/(2*A)} ${(-B - D)/(2*A)}`);
    else console.log(`Equation have no solutions.`);
});

File: basics_of_computer_algorithms/chapter_one/EX1_1/cpp/EXT1_1.cpp
#include <iostream>

int main()
{
    double A, B, C, D;

    std::cin >> A >> B >> C;

    D =  B*B - 4*A*C;
    std::cout << D;
    std::cout << std::endl;
    if( D == 0) {
        std::cout   << "Equation have one solution: " 
                    << -B/(2*A) << std::endl;
    }
    else if ( D > 0) {
        std::cout   << "Equation have two solutions: "
                    << (-B + D)/(2*A) << " " 
                    <<(-B - D)/(2*A) << std::endl;
    }
    else {
        std::cout   << "Equation have no solutions." << std::endl;
    };
    std::cin.ignore();

    return 0;
}